linode/linode_instance_helpers.go

```diff
@@ -717,10 +717,17 @@ func changeInstanceType(client *linodego.Client, instance *linodego.Instance, ta
 
 func changeInstanceDiskSize(client *linodego.Client, instance linodego.Instance, disk linodego.InstanceDisk, targetSize int, d *schema.ResourceData) error {
 	if instance.Specs.Disk > targetSize {
-		client.ResizeInstanceDisk(context.Background(), instance.ID, disk.ID, targetSize)
+		// This is idempotent, so we don't need to check the status of the instance.
+		client.ShutdownInstance(context.Background(), instance.ID)
```

Comment:
Resizing a disk that is not connected to the loaded config shouldn't require a shutdown.  Does the API distinguish between the two states or does it always return an error?

Reply:
This was left in the air back in https://github.com/displague/deprecated-terraform-provider-linode/pull/23#issuecomment-423197413 :-(

> Additional tests may be needed to verify that a resize on a live / running disk results in a shutdown and before the resize and a reboot afterwards.

Reply:
> Resizing a disk that is not connected to the loaded config shouldn't require a shutdown. Does the API distinguish between the two states or does it always return an error?

The API does not distinguish between disks that are attached / detached to the running instance via a config and will not permit resizing of an unattached disk associated with a running Linode.